content/utilities/PlanetOverview/plugin.js

```diff
@@ -0,0 +1,301 @@
+    // PlanetType Default value
+    this.planetType = PlanetType.SILVER_MINE;
+    // PlanetLevel Default value
+    this.minLevel = PlanetLevel.TWO;
+    // Slider Default value
+    this.topX = 10;
+    // checkbox MyPlanets
+    this.table = document.createElement("table");
+    this.table.style.width = "100%";
+    this.table.style.borderCollapse = "separate";
+    this.table.style.borderSpacing = "10px 0";
+    // default static text header of table
+    this.thead = document.createElement("thead");
+    this.thead.innerHTML = `<tr><th></th><th>Name</th><th>Level</th><th>Silver</th><th>SilverTime</th><th>Energy</th><th>EnergyTime</th>`;
+    this.table.appendChild(this.thead);
+    // default dynamic table body
+    this.tbody = document.createElement("tbody");
+    this.table.appendChild(this.tbody);
+  }
+  // Function to renderPlanets with sorting from home planet of preselected planet
+  renderPlanets(minLevel, planetType, middle, MyPlanets) {
+    // clean table body
+    this.tbody.innerHTML = null;
+    const selectedPlanetCoords = ui.selectedPlanet?.location?.coords;
+    const knownPlanets = [];
+    for (const planet of df.getAllPlanets()) {
+      if (planet.planetType != this.planetType) continue;
+      if (planet.planetLevel < this.minLevel) continue;
+      if (!planet?.location?.coords) continue;
+      if (myPlanets.checked && planet.owner != df.account) continue;
+
+      if (!selectedPlanetCoords) {
+        middle = df.getHomeCoords();
+      } else {
+        middle = selectedPlanetCoords;
+      }
+
+      knownPlanets.push({
+        locationId: planet.locationId,
+        distance: Math.floor(df.getDistCoords(planet.location.coords, middle)),
+      });
+    }
+
+    const sortedPlanets = knownPlanets
+      .sort((a, b) => a.distance - b.distance)
+      .slice(0, this.topX);
+    for (const [idx, p] of sortedPlanets.entries()) {
+      const planet = df.getPlanetWithId(p.locationId);
+      if (!planet) {
+        console.log(`Where is planet: ${p.locationId}`);
+        continue;
+      }
+      const row = document.createElement("tr");
+      row.style.color = getOwnerColor(planet);
+      row.onclick = () => {
+        ui.centerLocationId(planet.locationId);
+      };
+      // Create a Row per one planet
+      const silverPercent = Math.round(
+        100 / (planet.silverCap / Math.round(getSilver(planet)))
+      );
+      const fullSilverDate = new Date(
+        df.getSilverCurveAtPercent(planet, 99) * 1000
+      ).toString();
+      const fullSilverTime = [
+        fullSilverDate.substr(4, 6),
+        fullSilverDate.substr(16, 8),
+      ];
+
+      const energyPercent = Math.round(
+        100 / (planet.energyCap / Math.round(planet.energy))
+      );
+      const fullEnergyDate = new Date(
+        df.getEnergyCurveAtPercent(planet, 99) * 1000
+      ).toString();
+      const fullEnergyTime = [
+        fullEnergyDate.substr(4, 6),
+        fullEnergyDate.substr(16, 8),
+      ];
```

Comment:
Sometimes fullEnergyDate will be "Invalid Date", than fullEnergyTime will become "ild Da" because you make it a substr.
The same is true with the silver.
To fix this we could do something like this:
```
      const fullEnergyDate = new Date(
        df.getEnergyCurveAtPercent(planet, 99) * 1000
      ).toString();
      let fullEnergyTime = "";
      if (fullEnergyDate != "Invalid Date")
        fullEnergyTime = [ fullEnergyDate.substr(4, 6), fullEnergyDate.substr(16, 8), ];
```
This way it will be an empty str "" if the date is not valid and the energy is full, the same could be done for the silver

Reply:
resolved